client/settings/stripe-auth-account/account-status-panel.js

```diff
@@ -0,0 +1,182 @@
+import { __ } from '@wordpress/i18n';
+import { Icon } from '@wordpress/components';
+import { help } from '@wordpress/icons';
+import Chip from 'wcstripe/components/chip';
+import Tooltip from 'wcstripe/components/tooltip';
+import {
+	useAccountKeys,
+	useAccountKeysTestWebhookSecret,
+	useAccountKeysWebhookSecret,
+	useAccountKeysWebhookURL,
+	useAccountKeysTestWebhookURL,
+} from 'wcstripe/data/account-keys';
+import { useAccount } from 'wcstripe/data/account';
+
+/**
+ * Generates a status chip.
+ *
+ * @param {Object} props           The component props.
+ * @param {string} props.label     The label/type of the status.
+ * @param {string} props.text      The text of the status. The actual status message. eg "Enabled".
+ * @param {string} props.color     The color of the status.
+ * @param {JSX.Element} props.icon The icon of the status.
+ *
+ * @return {JSX.Element} The status chip.
+ */
+const Status = ( { label, text, color, icon } ) => {
+	return (
+		<div className="wcstripe-status">
+			<span className="wcstripe-status-text">{ label }</span>
+			<Chip text={ text } color={ color } icon={ icon } />
+		</div>
+	);
+};
+
+/**
+ * Generates an unconnected account tip.
+ *
+ * The tip is displayed when the account is incomplete and explains that the account is set up but not connected to
+ * the WooCommerce Stripe App.
+ *
+ * @return {JSX.Element} The unauthenticated account tip.
+ */
+const unconnectedAccountTip = () => {
+	return (
+		<Tooltip
+			content={ __(
+				'Your store has Stripe Account keys, however, it is not connected to WooCommerce.',
+				'woocommerce-gateway-stripe'
+			) }
+		>
+			<span>
+				<Icon icon={ help } size="16" />
+			</span>
+		</Tooltip>
+	);
+};
+
+/**
+ * Gets the account status.
+ *
+ * An account can be in one of three states:
+ *   - connected: The account is connected to WooCommerce.
+ *   - unconnected: There are account keys stored, but the account isn't connected to the WooCommerce Stripe App.
+ *   - disconnected: The account is not set up.
+ *
+ * @param {Object} accountKeys The account keys.
+ * @param {boolean} testMode   Whether the component is for test mode.
+ *
+ * @return {Object} The account status. Contains the status text (label), color, and optionally an icon.
+ */
+const getAccountStatus = ( accountKeys, testMode ) => {
+	const secretKey = testMode
+		? accountKeys.test_secret_key
+		: accountKeys.secret_key;
+	const publishableKey = testMode
+		? accountKeys.test_publishable_key
+		: accountKeys.publishable_key;
+
+	const accountStatus =
+		secretKey && publishableKey ? 'unconnected' : 'disconnected';
+
+	const accountStatusMap = {
+		connected: {
+			text: __( 'Connected', 'woocommerce-gateway-stripe' ),
+			color: 'green',
+		},
```

Comment:
Just noting because it came up during review that this `connected` case is currently not obtainable. It will be supported in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3290